scripts/tools/zap_regen_all.py

```diff
@@ -494,8 +494,6 @@ def main():
 
     timings = []
     if args.parallel:
-        # Ensure each zap run is independent
-        os.environ['ZAP_TEMPSTATE'] = '1'
```

Comment:
This has nothing to do with "stateful helpers" and everything to do with where ZAP stores its state like the sqlite DB: `~/.zap` or a temporary directory.  We're running things in parallel here, so if they all use `~/.zap` you will get random failures as they stomp on each other.

Reply:
You are correct that this can be potentially dangerous and technically doesn't have to do anything with stateful helpers. 

However, this is how zap was designed to operate. There is ONE database on the system. You load in the Matter XML files ONCE. You ideally regenerate everything you need to regenerate from ONE zap process, by passing in the `sdk.json` or a bunch of `zap` files on the CLI.

Zap also runs as a server, which serves both individual CLI requests for generation, and any UI HTTP requests. So when you do `zap do-something` on the command line, this makes `zap` realize that there is server already running (or it starts one if not), and then just tell that server to generate.

That makes for vast performance enhancements and so on.

It's really up to you if you want to remove ZAP_TEMPSTATE now or later. I don't particularly care.
What I think needs to happen though, is this:
   1.) The `sdk.json` file needs to be properly updated with the current state of what `zap_regen_all.py` does.
   2.) Then the `zap_regen_all.py` can simply be changed to a one-liner: `zap regenerateSdk --sdk sdk.json`.
   
After we do that, the ZAP_TEMPSTATE will not matter at all, as there will be a single process of zap regenerating everything anyway.

But I will not deny you that these changes definitely do have a potential to make CI a bit sad here and there until we work out the issues....

Reply:
What is creating the individual sql files for each application when running this script? I thought this setting was introduced for the state avoidance during generation. I also thought there would be a huge generation diff before the stateless changes without this setting since the statefulness would just pick up the first application it generates and thus giving a hug regen diff.

Reply:
> You ideally regenerate everything you need to regenerate from ONE zap process

That would perhaps be ideal, but we are not doing that.  We are doing a bunch of ZAP process runs, with different .json files, different .zap files, etc.

>  Then the zap_regen_all.py can simply be changed to a one-liner: zap regenerateSdk --sdk sdk.json.

zap_regen_all can regenerate multiple different sets of things depending on how it's run.

> After we do that, the ZAP_TEMPSTATE will not matter at all

That's plausible, but we are not there yet, so we can't remove it so far.

> What is creating the individual sql files for each application when running this script?

I'm not sure what you are asking.  Again, this script runs multiple `zap` instances in parallel.  Each instance tries to create the sql files, racing on them.  Then bad things happen.

It's not even that the sql files are different and you get regeneration diffs.  It's that you get corrupt files and insert statements start failing.

Reply:
All that said, while there are still mentions of ZAP_TEMPSTATE in the ZAP repo documentation, I don't see it being _used_ anywhere.  What seems to be used instead is the `--tempState` command-line option?